ui/app/components/pages/create-account/connect-hardware/connect-screen.js

```diff
@@ -0,0 +1,70 @@
+const { Component } = require('react')
+const PropTypes = require('prop-types')
+const h = require('react-hyperscript')
+
+class ConnectScreen extends Component {
+    constructor (props, context) {
+        super(props)
+    }
+
+    renderUnsupportedBrowser () {
+        return (
+            h('div', {}, [
+                h('div.hw-connect', [
+                    h('h3.hw-connect__title', {}, this.context.t('browserNotSupported')),
+                    h('p.hw-connect__msg', {}, this.context.t('chromeRequiredForTrezor')),
+                ]),
+                h(
+                    'button.btn-primary.btn--large',
+                    { onClick: () => global.platform.openWindow({
+                    url: 'https://google.com/chrome',
+                    }), style: { margin: 12 } },
+                    this.context.t('downloadGoogleChrome')
+                ),
+            ])
+        )
+    }
+
+    renderConnectScreen () {
+        return (
+            h('div', {}, [
+                h('div.hw-connect', [
+                    h('h3.hw-connect__title', {}, this.context.t('trezorHardwareWallet')),
+                    h('p.hw-connect__msg', {}, this.context.t('connectToTrezorHelp')),
+                    h('p.hw-connect__msg', {}, [
+                        this.context.t('connectToTrezorTrouble'),
+                        h('a.hw-connect__link', {
+                            href: 'https://support.metamask.io/',
+                            target: '_blank',
+                        }, ` ${this.context.t('learnMore')}`),
+                    ]),
+                ]),
+                h(
+                    'button.btn-primary.btn--large',
+                    { onClick: this.props.connectToTrezor.bind(this) },
+                    this.props.btnText
+                ),
+            ])
+        )
+      }
+
+    render () {
+        const isChrome = window.navigator.userAgent.search('Chrome') !== -1
```

Comment:
i would recommend checking for `window.chrome` (necessary for trezor connect afaict: https://github.com/trezor/connect/blob/v4/connect.js#L27) instead of checking UA here. Brave may change our UA to not include Chrome in the near future, but we do support window.chrome API

Reply:
That's a good idea! Anyway, I've to revisit the whole browser thing cause I had the (wrong) assumption that this would work only in Chrome, but I tried earlier today in Firefox and it works.

Reply:
@diracdeltas  I just figured out what's going on...

This example https://connect.trezor.io/4/examples/signtx-ethereum.html works fine on firefox.
If I try to call any function that will open the trezor popup from the extension, in firefox I get a JS error: "Window Blocked"

I've tried to whitelist the extension URL in the popup blocking settings and also disabled the popup blocker completely with no luck.

Based on this, I assume that extensions are not allowed to open popups using window.open, which is what Trezor Connect is using (https://github.com/trezor/connect/blob/v4/connect.js#L902)

I'll reach out to Trezor about this but for now I'll have to disable firefox support and test in other browsers to see which ones will support this feature.